src/algoliasearch.helper.js

```diff
@@ -163,22 +176,34 @@ AlgoliaSearchHelper.prototype.getQuery = function() {
 AlgoliaSearchHelper.prototype.searchOnce = function(options, cb) {
   var tempState = !options ? this.state : this.state.setQueryParameters(options);
   var queries = requestBuilder._getQueries(tempState.index, tempState);
+  var self = this;
+
+  this._currentNbQueries++;
+
   if (cb) {
     return this.client.search(
       queries,
       function(err, content) {
+        self._currentNbQueries--;
+        if (self._currentNbQueries === 0) self.emit('noMoreSearch');
```

Comment:
Let's go for `searchQueueEmpty` as mentionned during lunch :) Ok for you, @Haroenv ?